mrp_unbuild_cust_alu_analytics/models/mrp_unbuild.py

```diff
@@ -0,0 +1,187 @@
+# © 2024 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError, UserError
+import os
+from lxml import etree
+from io import StringIO, BytesIO
+import logging
+
+
+class MrpUnbuild(models.Model):
+    _inherit = "mrp.unbuild"
+
+    process_type_id = fields.Many2one('mrp.unbuild.process.type', string='Process type')
+    incidence_ids = fields.One2many(
+        comodel_name="mrp.unbuild.incidence",
+        inverse_name="unbuild_id",
+        string="Incidences",
+    )
+    analytic_ids = fields.Many2many(
+        comodel_name="qc.inspection",
+        string="Analytics",
+        compute="compute_move_analytics",
+        store=False
+    )
+
+    def compute_move_analytics(self):
+        for record in self:
+            analytics = self.env["qc.inspection"].search([("unbuild_id", "=", record.id)])
+            record.analytic_ids = analytics
+
+    @api.onchange('product_id')
+    def _onchange_product_id(self):
+        if self.product_id and self.process_type_id:
+            self.bom_id = self.product_id.product_tmpl_id.bom_ids.filtered(lambda r: r.process_type_id == self.process_type_id)
+        else:
+            super(MrpUnbuild, self)._onchange_product_id()
+
+    @api.onchange('bom_id')
+    def _onchange_bom_id(self):
+        if self.bom_id:
+            self.product_uom_id = self.bom_id.product_uom_id.id
+        else:
+            self.product_uom_id = False
+
+    @api.constrains('bom_quants_ids')
+    def _contrains_bom_quants_ids(self):
+        total_product_qty = sum(self.bom_quants_total_ids.mapped('total_qty'))
+        self.product_qty = total_product_qty if total_product_qty > 0 else 1
+
+    @api.constrains("incidence_ids")
+    def _constrains_incidences_ids(self):
+        if self.incidence_ids:
+            self.shift_stop_time = sum(self.incidence_ids.mapped('duration'))
+        else:
+            self.shift_stop_time = 0
+
+    @api.onchange("product_id", "process_type_id")
+    def _onchange_product_id_process_type_id(self):
+        for record in self:
+            domain_bom_id = [
+                "&",
+                ("process_type_id", "=", record.process_type_id.id),
+                ("product_tmpl_id", "=", record.product_id.product_tmpl_id.id)
+            ]
+            domain_product_id = [
+                ("process_type_ids", "=", record.process_type_id.id),
+            ]
+        return {"domain": {"bom_id": domain_bom_id, "product_id": domain_product_id}}
+
+    def read_qc_inspection_xml(self):
```

Comment:
Better name (this is a protected method, only called by a cron):
```suggestion
    def _read_qc_inspection_xml(self):
```

Reply:
Pending to be answered/fixed

Reply:
Solved